python/ray/serve/config.py

```diff
@@ -120,618 +98,74 @@ def get_downscale_smoothing_factor(self) -> PositiveFloat:
-            ray_actor_options,
-            placement_group_bundles,
-            placement_group_strategy,
-            max_replicas_per_node,
-        )
-
-        config._deployment_def = deployment_def
-        config._init_args = init_args
-        config._init_kwargs = init_kwargs
-
-        return config
-
-    def _validate_ray_actor_options(self):
-        if not isinstance(self.ray_actor_options, dict):
-            raise TypeError(
-                f'Got invalid type "{type(self.ray_actor_options)}" for '
-                "ray_actor_options. Expected a dictionary."
-            )
-        # Please keep this in sync with the docstring for the ray_actor_options
-        # kwarg in api.py.
-        allowed_ray_actor_options = {
-            # Resource options
-            "accelerator_type",
-            "memory",
-            "num_cpus",
-            "num_gpus",
-            "object_store_memory",
-            "resources",
-            # Other options
-            "runtime_env",
-        }
-
-        for option in self.ray_actor_options:
-            if option not in allowed_ray_actor_options:
-                raise ValueError(
-                    f"Specifying '{option}' in ray_actor_options is not allowed. "
-                    f"Allowed options: {allowed_ray_actor_options}"
-                )
-        ray_option_utils.validate_actor_options(self.ray_actor_options, in_options=True)
-
-        # Set Serve replica defaults
-        if self.ray_actor_options.get("num_cpus") is None:
-            self.ray_actor_options["num_cpus"] = 1
-
-    def _validate_max_replicas_per_node(self) -> None:
-        if self.max_replicas_per_node is None:
-            return
-        if not isinstance(self.max_replicas_per_node, int):
-            raise TypeError(
-                f"Get invalid type '{type(self.max_replicas_per_node)}' for "
-                "max_replicas_per_node. Expected None or an integer "
-                f"in the range of [1, {MAX_REPLICAS_PER_NODE_MAX_VALUE}]."
-            )
-        if (
-            self.max_replicas_per_node < 1
-            or self.max_replicas_per_node > MAX_REPLICAS_PER_NODE_MAX_VALUE
-        ):
-            raise ValueError(
-                f"Invalid max_replicas_per_node {self.max_replicas_per_node}. "
-                "Valid values are None or an integer "
-                f"in the range of [1, {MAX_REPLICAS_PER_NODE_MAX_VALUE}]."
-            )
-
-    def _validate_placement_group_options(self) -> None:
-        if (
-            self.placement_group_strategy is not None
-            and self.placement_group_strategy not in VALID_PLACEMENT_GROUP_STRATEGIES
-        ):
-            raise ValueError(
-                f"Invalid placement group strategy '{self.placement_group_strategy}'. "
-                f"Supported strategies are: {VALID_PLACEMENT_GROUP_STRATEGIES}."
-            )
+class ProxyLocation(str, Enum):
```

Comment:
Why is this marked as DeveloperAPI, if it's a public parameter in `serve.start`? (is there a plan to promote this to PublicAPI in the future?)

Reply:
fair point, let me just upgrade it